src/main/java/lesser/finalproject/RijksSearchFrame.java

```diff
@@ -0,0 +1,137 @@
+package lesser.finalproject;
+
+import hu.akarnokd.rxjava3.swing.SwingSchedulers;
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.disposables.CompositeDisposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import lesser.finalproject.json.ArtObject;
+import lesser.finalproject.json.CollectionResponse;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.io.IOException;
+import java.net.URL;
+
+public class RijksSearchFrame extends JFrame {
+
+    private JTextField searchField;
+    private JButton searchButton;
+    private JPanel resultsPanel;
+    private JButton nextPageButton;
+    private JButton prevPageButton;
+    private JLabel statusLabel;
+    private RijksService rijksService;
+    private int currentPage = 1;
+    private CompositeDisposable disposable = new CompositeDisposable();
+
+    public RijksSearchFrame() {
+        setTitle("Rijksmuseum Art Search");
+        setSize(800, 800);
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setLayout(new BorderLayout());
+
+        searchField = new JTextField(30);
+        searchButton = new JButton("Search");
+        resultsPanel = new JPanel(new GridLayout(0, 4, 10, 10));
```

Comment:
You could've still used GridBagLayout. I just wanted to know where you learned it from.